db/migrate/20220902115804_create_response_items.rb

```diff
@@ -0,0 +1,13 @@
+class CreateResponseItems < ActiveRecord::Migration[7.0]
+  def change
+    create_table :response_items do |t|
+      t.belongs_to :query, index: true, foreign_key: true
+      t.integer :source, null: false
+      t.integer :media_type, null: false
+      t.string :url, null: false
```

Comment:
We probably need more than one URL per response item. Commonly with NLI items, there's the URL of the *item* at the NLI, but we'd also be interested, and watch to store for quick fetching, the URL for the *media* itself (in the cause of audio, image, etc.), and/or for a thumbnail.

Do you think we should just add a couple of extra URL fields (media_url, thumbnail_url) or another has_many association linking an arbitrary number of URLs to a result item? The former would certainly be enough for the POC.

Reply:
I would prefer to go with first approach.
Second one looks redundant to me: if we're going to have data from different sources then we should expect different set of properties from each source. So I would propose to use most common set of properies for our data model. User can always go to source website and see more details if he need them.

Reply:
I've added two columns to response_items, see last commit